bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java

```diff
@@ -132,7 +135,17 @@ public void onError(final @Nullable SpeedTestError testError, final @Nullable St
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
             freeRefreshTask();
             return;
-        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError) || SpeedTestError.SOCKET_ERROR.equals(testError)
+        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError)) {
+            timeouts--;
+            if (timeouts <= 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Max timeout count reached");
```

Comment:
If it times-out there is no simple way to restart it? Maybe reset the timeouts value when the a ON command is passed to the handleCommand and in the onComplete restart the refresh task if it's not running?

Reply:
It's an idea but does Thing accept commands when they are OFFLINE ?

Reply:
Yes based on the Javadoc of the handleCommand method it should be called:
```
This method is only called, if the thing has been initialized (status ONLINE/OFFLINE/UNKNOWN).
```